### DISCUSSION TITLE
Can I get the AuthenticationBackend to set/remove multiple cookies on login/logout?

### DISCUSSION THREAD

**Discussion by sedrik** [fastapi-users/fastapi-users#1346] [General]
Hi

We are using the CookieTransport for our jwt's and it is working nicely. Is there a good way to set and remove a companion cookie that is not HTTPOnly that can be used by the frontend to validate if we actually have access to the credential cookie or not.

While we can track the state manually in the frontend it is very easy for it to diverge from the validity of the cookie (either expiration or if the logout calls fails for whatever reason).

**Comment by frankie567** [comment]
I think you can achieve this by overloading the [`CookieTransport` class](https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/authentication/transport/cookie.py). Something like:

```py


class MyCookieTransport(CookieTransport):
    def _set_login_cookie(self, response: Response, token: str) -> Response:
        response = super()._set_login_cookie(response, token)
        response.set_cookie("is_authed", "true", httponly=False)
        return response

    def _set_logout_cookie(self, response: Response) -> Response:
        response = super()._set_logout_cookie(response)
        response.set_cookie("is_authed", "", httponly=False)
        return response

```

**Comment by sedrik** [comment]
Thanks I will try it out and let you know how it goes :)

**Comment by sedrik** [comment]
I took it for a spin and the approach works nicely.

Thanks :)

**Comment by sedrik** [comment]
Would you like me to document this in a PR? :)

**Reply by frankie567** [reply]
Could be a nice addition to our **Cookbook** section, yes 😄
